### ISSUE TITLE
[Bug] Qwen3.6: GDN per-slot state is not remapped on host-sampling decodes

### ISSUE THREAD

**Issue by viktorpusTT** [tenstorrent/tt-metal#51982]
### Summary

Qwen3.6's per-slot GDN recurrent/conv state is remapped only when the decode step
samples on device. vLLM delivers `slot_remap` on device-sampling decodes only
(pre-contract behaviour), so a batch-condense that happens on a host-sampling
decode leaves GDN state attached to the previous occupant of the slot.

Verified on `main` at 74d9b26577a.

### Where

`models/demos/blackhole/qwen36/tt/qwen36_vllm.py:290-294`

```python
if model.num_devices > 1 and model.args.max_batch_size > 1:
    slot_remap = kwargs.get("slot_remap")
    if slot_remap is not None:
        model._remap_gdn_slots(slot_remap)
return super().decode_forward(*args, **kwargs)
```

`kwargs.get("slot_remap")` returns `None` on any host-sampling decode, because
the vLLM plugin only passes the keyword when `perform_device_sampling` is true.

### Why it is reachable

This does not need async scheduling — `supports_async_decode` is `False` for this
model (`:51-55`). It needs only a step that falls back to host sampling while the
batch layout changes. vLLM falls back whenever a request in the batch uses
structured output, penalties, logprobs, `bad_words`, or a logits processor.
`InputBatch.condense` runs on any mid-batch removal, so with `max_batch_size > 1`
and `num_devices > 1` (TP batched serving) the combination is ordinary traffic,
not an edge case.

The consequence is that the next forward reads slot *i*'s recurrent and
convolution state while it belongs to a request that has moved or finished. The
symptom is silent output corruption for the affected slot, not a crash.

### Fix options

1. **Preferred:** advertise `decode_input_update_contract = 1`. The paired vLLM
   change delivers `slot_remap` on every version-1 decode in both sampling modes,
   which fixes this without further adapter work beyond honouring the four
   commands. See tenstorrent/vllm#456 and its follow-up.
2. If v1 adoption is further out, note that `_remap_gdn_slots` is already
   unconditional on sampling mode *within* this adapter — the gap is purely that
   vLLM withholds the argument. There is no adapter-side workaround short of the
   contract, so option 1 is effectively required.

Note the docstring in `models/demos/blackhole/qwen36/tt/gdn/tp.py:783-784`
already records the underlying asymmetry ("which the plugin's slot_remap does not
itself move").

### Related

- tenstorrent/vllm#449 — failure mode 1 in the decode reload contract:
  "vLLM does not deliver the remap on a host-sampling decode"
- No adapter on `main` advertises `decode_input_update_contract` yet, so the vLLM
  fix is inert until this one opts in

**Comment by atupe-tt** [comment]
@viktorpusTT Thanks for reporting this bug! I'll fix this in my current [PR](https://github.com/tenstorrent/tt-metal/pull/51556), after this [vLLM PR](https://github.com/tenstorrent/vllm/pull/458) gets merged (as it is a continuation of https://github.com/tenstorrent/vllm/pull/456), since the changes are dependent on it.

cc: @mbahnasTT @ssinghalTT
